docs/guides/using-org-slugs.mdx

```diff
@@ -0,0 +1,397 @@
+    },
+    {
+      organizationSyncOptions: {
+        organizationPatterns: ['/orgs/:slug', '/orgs/:slug/(.*)'],
+        personalAccountPatterns: ['/me', '/me/(.*)'],
+      },
+    },
+  )
+
+  export const config = {
+    matcher: [
+      // Skip Next.js internals and all static files, unless found in search params
+      '/((?!_next|[^?]*\\.(?:html?|css|js(?!on)|jpe?g|webp|png|gif|svg|ttf|woff2?|ico|csv|docx?|xlsx?|zip|webmanifest)).*)',
+      // Always run for API routes
+      '/(api|trpc)(.*)',
+    ],
+  }
+  ```
+
+  > [!TIP]
+  > If you prefer to have an [organization ID](/docs/references/javascript/organization/organization#properties)
+  > appear in your URL rather than slugs, you can use `:id` placeholder instead of `:slug` in both `organizationPatterns`
+  > and `personalAccountPatterns`.
+
+  ### Server Rendered Organization Page
+
+  Now that clerkMiddleware is configured to activate organizations, you can build an organization-specific home page.
+
+  First, you must handle the case where the middleware cannot activate the organization based on the URL. This can
+  happen if no organization with the given slug exists, or if the given user isn't a member of the organization.
+
+  When this happens, the middleware will not change the active organization, so whatever organization was previously
+  active will remain active. To help with troubleshooting, a message will also be emitted by the middleware to the server
+  log:
+
+  > Clerk: Organization activation handshake loop detected. This is likely due to an invalid organization ID or slug.
+  > Skipping organization activation.
+
+  > [!CAUTION]
+  > It is ultimately the responsibility of the page to ensure that it renders the appropriate content for a given URL, and
+  > to handle the case where the expected organization is not active. In this case, we detect the actual organization slug
+  > as a NextJS [Dynamic Route](https://nextjs.org/docs/pages/building-your-application/routing/dynamic-routes) param, and
+  > compare it to the active organization slug. If they don't match, we render an error message and the [Organization List](/docs/components/organization/organization-list)
+  > component to allow the user to select a valid organization.
+
+  ```tsx {{ filename: 'src/app/orgs/[slug]/page.tsx' }}
+  import { auth } from '@clerk/nextjs/server'
+  import { OrganizationList } from '@clerk/nextjs'
+
+  export default function Home({ params }: { params: { slug: string } }) {
+    const authObject = auth()
+    const orgSlug = authObject.orgSlug
+
+    if (params.slug != orgSlug) {
+      return (
+        <>
+          <p>Sorry, organization {params.slug} is not valid.</p>
+          <OrganizationList
+            hidePersonal={false}
+            hideSlug={false}
+            afterCreateOrganizationUrl="/orgs/:slug"
+            afterSelectOrganizationUrl="/orgs/:slug"
+            afterSelectPersonalUrl="/me"
+          />
+        </>
+      )
+    }
```

Comment:
```suggestion
  export default function Home({ params }: { params: { slug: string } }) {
    const { orgSlug } = auth()
   
    // Verify mismatch between route param and active organization slug from user's session 
    if (params.slug != orgSlug) {
      return (
        <>
          <p>Sorry, organization {params.slug} is not valid.</p>
          <OrganizationList
            hidePersonal={false}
            hideSlug={false}
            afterCreateOrganizationUrl="/orgs/:slug"
            afterSelectOrganizationUrl="/orgs/:slug"
            afterSelectPersonalUrl="/me"
          />
        </>
      )
    }
```

Another suggestion would also be to return a https://nextjs.org/docs/app/api-reference/functions/not-found